README.md
# zf_flutter_kit

A flutter suite library containing common components(button,text,dialog,slider,route page etc.) for easy use.

# Instructions for use
For more,see example project.

# Installing
Use this package as a library
1. Depend on it
Add this to your package's pubspec.yaml file:
```yaml
dependencies:
  zf_flutter_kit: ^0.0.2
```

2. Install it
You can install packages from the command line:

with Flutter:


```yaml
flutter pub get
```

Alternatively, your editor might support ```flutter pub get```. Check the docs for your editor to learn more.

3. Import it
Now in your Dart code, you can use:


```dart
import 'package:zf_flutter_kit/zf_flutter_kit.dart';
```

# Screenshot

## Widget
### ZFButton
```dart
    ZFButton(
      {this.text = '',
      this.onPressed,
      this.borderRadius = 8,
      this.borderWidth = 1,
      this.color = Colors.white,
      this.highlightColor = _highlightColor,
      this.textColor = _color,
      this.highlightTextColor = Colors.white,
      this.borderColor = _color,
      this.highlightBorderColor = _highlightColor,
      this.textStyle,
      this.selected = false,
      });
```
### Text、BaseWidget（a simple PageRoute）、Scrollview、slider etc.

![widget](https://raw.githubusercontent.com/jtans/zf_flutter_kit/master/screenshot/Simulator%20Screen%20Shot%20-%20iPhone%2011%20Pro%20Max%20-%202020-03-31%20at%2015.02.59.png)

## Dialog
```dart
ZFAlertDialog.show(context, 'Your Title', 'Your Message',
                onOkPressed: () {
              return true; //return true to close, when press ok button
            });
```
![image](https://raw.githubusercontent.com/jtans/zf_flutter_kit/master/screenshot/Simulator%20Screen%20Shot%20-%20iPhone%2011%20Pro%20Max%20-%202020-03-31%20at%2015.03.06.png)
## custom dialog
```dart
ZFAlertDialog.show(context, 'Your Title', 'Your Message',
                onOkPressed: () {
              return true;
            }, footerContent: () {
              return Container(
                color: Colors.red,
                width: 200,
                height: 44,
              );
            }());
```
![image](https://raw.githubusercontent.com/jtans/zf_flutter_kit/master/screenshot/Simulator%20Screen%20Shot%20-%20iPhone%2011%20Pro%20Max%20-%202020-03-31%20at%2015.03.11.png)

## custom dialog content
```dart
showDialog(
                context: context,
                builder: (ctx) {
                  return ZFDialog(
                    cancelText: 'Cancel',
                    okText: 'Ok',
                    content: () {
                      return Image.network(
                          'https://www.baidu.com/img/bd_logo1.png');
                    }(),
                  );
                });
```
![image](https://raw.githubusercontent.com/jtans/zf_flutter_kit/master/screenshot/Simulator%20Screen%20Shot%20-%20iPhone%2011%20Pro%20Max%20-%202020-03-31%20at%2015.03.16.png)

## Navigator
```dart
//push open a page
Navigator.of(context).zfpush(MinePage());

//push open a webview page
Navigator.of(context)
                .zfopenWebView('https://github.com', 'Your web page title');
```
![image](https://raw.githubusercontent.com/jtans/zf_flutter_kit/master/screenshot/push.gif)



CHANGELOG.md
## [0.0.1] - 2020/3/31.

* first version: button,text,dialog,webview,navigator component

## [0.0.2] - 2020/3/31.

* edit readme.md

## [0.0.3] - 2020/4/1.

* add roate widget

example/lib/main.dart
import 'package:example/rotate_page.dart';
import 'package:flutter/material.dart';
import 'mine_page.dart';
import 'package:zf_flutter_kit/zf_flutter_kit.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends ZFBaseWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends ZFBaseWidgetState {
  @override
  String get title => 'Home';

  Widget _buildDivider(String title) {
    var t = title.padLeft(20, '-') + ''.padRight(20, '-');
    return ZFThemeText(
      text: t,
    );
  }

  @override
  Widget buildBody(BuildContext context) {
    return Column(
      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
      children: <Widget>[
        _buildDivider('Button'),
        ZFButton(
          text: 'Button',
        ),
        ZFButton(
          borderColor: Colors.red,
          borderRadius: 15,
          text: 'Custom Button',
        ),
        ZFThemeButton(
          text: 'Theme Button',
        ),
        ZFThemeText(
          text: 'Theme Text',
        ),
        ZFThemeButton(
          text: 'Rotate Widget',
          onPressed: (){
            Navigator.of(context).zfpush(RotatePage());
          },
        ),
        _buildDivider('Dialog'),
        ZFButton(
          text: 'Show Alert Dialog',
          onPressed: () {
            ZFAlertDialog.show(context, 'Your Title', 'Your Message',
                onOkPressed: () {
              return true; //return true to close, when press ok button
            });
          },
        ),
        ZFButton(
          text: 'Show Custom Alert Dialog',
          onPressed: () {
            ZFAlertDialog.show(context, 'Your Title', 'Your Message',
                onOkPressed: () {
              return true;
            }, footerContent: () {
              return Container(
                color: Colors.red,
                width: 200,
                height: 44,
              );
            }());
          },
        ),
        ZFButton(
          text: 'Show Custom Dialog',
          onPressed: () {
            showDialog(
                context: context,
                builder: (ctx) {
                  return ZFDialog(
                    cancelText: 'Cancel',
                    okText: 'Ok',
                    content: () {
                      return Image.network(
                          'https://www.baidu.com/img/bd_logo1.png');
                    }(),
                  );
                });
          },
        ),
        _buildDivider('Navigator'),
        ZFButton(
          text: 'push a page',
          onPressed: () {
            Navigator.of(context).zfpush(MinePage());
          },
        ),
        ZFButton(
          text: 'push open a webview',
          onPressed: () {
            Navigator.of(context)
                .zfopenWebView('https://github.com', 'Your web page title');
          },
        )
      ],
    );
  }
}
